### ISSUE TITLE
1.10 Icinga PowerShell Service is not listening on 5668

### ISSUE THREAD

**Issue by puseidr** [Icinga/icinga-powershell-framework#586]
In the past i had trouble with high CPU usage on my Windows VMs using the icinga Powershell Framework and Plugins
after some positive testings with the API Check Forwarder Feature i updated my environment (all of them are Windows Server 2019 Datacenter) to the newest version of the framework and plugins and enabled the API Check Forwarder feature with the following commands:

```
Install-IcingaComponent -Name service -Confirm -Force;
Register-IcingaBackgroundDaemon -Command 'Start-IcingaWindowsRESTApi';
Add-IcingaRESTApiCommand -Command 'Invoke-IcingaCheck*' -Endpoint 'apichecks';
Enable-IcingaFrameworkApiChecks;
Restart-IcingaWindowsService;
```

on most of my VMs its working but there are some machines where the powershell service is not listening on port 5668 
i'm checking the port with this command:
```
netstat -aon | ForEach-Object { if ($_ -like '*5668*') { $_ } }
```
most of the non working machines are showing nothing and some of them do show this:
```
 netstat -aon | ForEach-Object { if ($_ -like '*5668*') { $_ } }
  TCP    127.0.0.1:53275        127.0.0.1:5668         SYN_GESENDET    34012
```
and when i check the URL "https://localhost:5668/v1/" with the browser i get no response :(

some of the problem machines do even show this error while installing the powershell service:
```
Install-IcingaComponent -Name service -Confirm -Force;
[Notice]: Downloading "service" from "//my-local-repo/icingastable/service/icinga-service-v1.2.0.zip"
[Notice]: Installing component "service" into "C:\Program Files\icinga-framework-service"
[Notice]: Updating Icinga PowerShell Service binary
[Notice]: Service User ".\LocalSystem" for service "icingapowershell" successfully updated
[Notice]: Stopping service "icingapowershell"
[Notice]: Restarting service "icingapowershell"
[Notice]: Stopping service "icingapowershell"
[Error]: Failed to update the service user: [SC] ChangeServiceConfig FEHLER 1057:Der Kontoname ist unzulässig oder nicht vorhanden, oder das Kennwort ist für den angegebenen Kontennamen ungültig.
[Notice]: Installation of component "service" was successful
```
at the end the service is running as "LocalSystem" but even running as another administrator user it is not listening on port 5668

Any ideas to fix this?
BTW: does the icingapowershell service write a logfile? (didn't find one)

**Comment by t3easy** [comment]
I've the same problem on a new server with a fresh IFW 1.10.0 installation, not updated from previous version. Nothing listens on port 5668.

My install with default JEA user.

**Comment by t3easy** [comment]
@puseidr The only log I know of is in the Windows Eventlog.
Under Applications and Services > Icinga for Windows

There I get lots of the errors 
source: IfW::Framework
ID: 1553

On the fresh server.

Servers upgrades from 1.9.2 are fine, just checked.

**Comment by t3easy** [comment]
I'm happy to provide additional information, logs, config, etc. if you tell me what I should provide. @LordHepipud could you assist?

**Comment by Philipp4292** [comment]
Can confirm. I crashed the 5668 listener with the following command:
`Set-IcingaServiceUser -User 'NT AUTHORITY\SYSTEM' -SetPermission`

Reinstalled everything from scratch. No Listener.

This command fixed it:
`Install-IcingaComponent -Name service -Confirm -Force;`

**Comment by t3easy** [comment]
I can't confirm the service force install workaround.
Maybe because we use JEA...

Even after installing the service again with
```powershell
PS C:\Users\Administrator> Install-IcingaComponent -Name service -Confirm -Force;
[Notice]: Downloading "service" from "https://packages.icinga.com/IcingaForWindows/stable/service/icinga-service-v1.2.0.zip"
[Notice]: Installing component "service" into "C:\Program Files\icinga-framework-service"
[Notice]: Updating Icinga PowerShell Service binary
[Notice]: Stopping Icinga PowerShell service
[Notice]: Stopping service "icingapowershell"
[Notice]: Service User ".\LocalSystem" for service "icingapowershell" successfully updated
[Notice]: Stopping service "icingapowershell"
[Notice]: Restarting service "icingapowershell"
[Notice]: Stopping service "icingapowershell"
[Notice]: Starting Icinga PowerShell service
[Notice]: Starting service "icingapowershell"
[Notice]: Service User "SERVER\icinga" for service "icingapowershell" successfully updated
[Notice]: User updated successfully.
[Notice]: Service User "SERVER\icinga" for service "icinga2" successfully updated
[Notice]: Service User "SERVER\icinga" for service "icingapowershell" successfully updated
[Notice]: Restarting service "icinga2"
[Notice]: Stopping service "icingapowershell"
[Notice]: Restarting service "icingapowershell"
[Notice]: Installation of component "service" was successful
```

Nothing listens on port 5668.

**Comment by LordHepipud** [comment]
I just played around with this and the only time I could reproduce the issue was by using JEA.
In some cases, it might happen that the `wsmprovhost.exe` process (inside which the PowerShell daemon is running in JEA) is not terminated properly, meaning that the service is neither properly dead nor properly available, causing issues.

I need to test with `LocalSystem`, but I would not recommend this in general and use JEA instead, as LocalSystem has way too many permissions and shouldn't be used for security reasons.

By using the `NetworkService` user, the API just started fine but couldn't apply many of the checks because if missing permissions,

For JEA, you have to ensure to update the JEA profile **after** you registered the service for the background daemon, because otherwise the command is not part of the profile.

**Comment by puseidr** [comment]
i've checked the problem on some of my affected systems.
``Install-IcingaComponent -Name service -Confirm -Force;`` didn't fixed anything :(

Even running the "icingapowershell" service as a different user (LocalSystem, a local administrator user and NetworkService) with and without JEA didn't solve the problem... after (re)starting the service its shortly listening (for 1 to 2 seconds ) but then stops listening

**Comment by Philipp4292** [comment]
The behavior is strange. Sometimes it runs for several days with no problems, then it's back for a day or two. The troubleshooting process begins again: Only solutions so far restart services, reinstall, cmd as mentioned above, something helps then.
Currently it only occurs on physical hardware.

**Comment by LordHepipud** [comment]
This should be resolved with v1.12.0, as we made some changes on the setup routines as well as ensured that certificates are on track and some other changes related to the JEA-Profile builder.

In case the API is not listening afterwards, you can run

```
Test-IcingaForWindows;
```

starting with v1.12.0, which should print a huge variety of reports and tests to ensure the environment is working properly. We also added some other checks and updated the knowledge base with known issues.
